### PR TITLE
Update Update-AzSqlDatabaseAdvancedThreatProtectionSetting to an updated version and allow storage account as an optional parameter

### PR DESCRIPTION
**Pull request by ayeletshpigelman**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by BethanyZhou** [comment]
Hi @ayeletshpigelman , please update the corresponding test cases of `Update-AzSqlDatabaseAdvancedThreatProtectionSetting` as well.

File: src/Sql/Sql/ThreatDetection/Services/SqlThreatDetectionAdapter.cs [hunk 1]
```diff
@@ -318,12 +318,31 @@ private Management.Sql.Models.ManagedServerSecurityAlertPolicy PolicizeManagedIn
         /// <summary>
         /// Takes the cmdlets model object and transform it to the policy as expected by the endpoint
         /// </summary>
-        private DatabaseSecurityAlertPolicyCreateOrUpdateParameters PolicizeDatabaseSecurityAlertModel(BaseThreatDetectionPolicyModel model, string storageEndpointSuffix)
+        private Management.Sql.Models.DatabaseSecurityAlertPolicy PolicizeDatabaseSecurityAlertModel(BaseThreatDetectionPolicyModel model, string storageEndpointSuffix)
```
Comment (tomerrt) [inline src/Sql/Sql/ThreatDetection/Services/SqlThreatDetectionAdapter.cs:321]: why not using/alias?

File: src/Sql/Sql/help/Update-AzSqlDatabaseAdvancedThreatProtectionSetting.md [hunk 2]
```diff
@@ -22,15 +22,14 @@ Update-AzSqlDatabaseAdvancedThreatProtectionSetting [-PassThru] [-NotificationRe
...
 
 ## EXAMPLES
 
 ### Example 1: Set the advanced threat protection settings for a database
 ```
-PS C:\>Update-AzSqlDatabaseAdvancedThreatProtectionSetting -ResourceGroupName "ResourceGroup11" -ServerName "Server01" -DatabaseName "Database01" -NotificationRecipientsEmails "admin01@contoso.com;secadmin@contoso.com" -EmailAdmins $False -ExcludedDetectionType "Sql_Injection_Vulnerability", "SQL_Injection" -StorageAccountName "mystorageAccount"
+PS C:\>Update-AzSqlDatabaseAdvancedThreatProtectionSetting -ResourceGroupName "ResourceGroup11" -ServerName "Server01" -DatabaseName "Database01"
 ```
 
 This command sets the advanced threat protection settings for a database named Database01 on the server named Server01.
```
Comment (tomerrt) [inline src/Sql/Sql/help/Update-AzSqlDatabaseAdvancedThreatProtectionSetting.md:35]: Also, lets edit -EmailAdmins, -ExcludedDetectionType, -NotificationRecipientsEmails, -RetentionInDays, -StorageAccountName with the following note:
"Leave empty to use the enhanced Azure Security Center experience for consuming alerts, configure notifications and handle suppressions."

**Review by tomerrt — CHANGES_REQUESTED** [review]

File: src/Sql/Sql/ThreatDetection/Services/SqlThreatDetectionAdapter.cs (hunk 1 above)
Comment (ayeletshpigelman) [reply src/Sql/Sql/ThreatDetection/Services/SqlThreatDetectionAdapter.cs:321]: fixed

File: src/Sql/Sql/help/Update-AzSqlDatabaseAdvancedThreatProtectionSetting.md (hunk 2 above)
Comment (ayeletshpigelman) [reply src/Sql/Sql/help/Update-AzSqlDatabaseAdvancedThreatProtectionSetting.md:35]: fixed

**Comment by ayeletshpigelman** [comment]
> Hi @ayeletshpigelman , please update the corresponding test cases of `Update-AzSqlDatabaseAdvancedThreatProtectionSetting` as well.

done.

**Comment by ayeletshpigelman** [comment]
@BethanyZhou  - I fixed the tests

**Review by BethanyZhou — APPROVED** [review]

**Pull request merged**